bgpd/bgp_bmp.c

```diff
@@ -314,11 +369,26 @@ static void bmp_put_info_tlv(struct stream *s, uint16_t type,
 	stream_put(s, string, len);
 }
 
+static void __attribute__((unused))
+bmp_put_vrftablename_info_tlv(struct stream *s, struct bmp *bmp)
+{
+
+#define BMP_INFO_TYPE_VRFTABLENAME 3
+	const char *vrftablename = "global";
+	if (bmp->targets->bgp->inst_type != BGP_INSTANCE_TYPE_DEFAULT) {
+		struct vrf *vrf = vrf_lookup_by_id(bmp->targets->bgp->vrf_id);
+
+		vrftablename = vrf ? vrf->name : NULL;
+	}
+	if (vrftablename != NULL)
+		bmp_put_info_tlv(s, BMP_INFO_TYPE_VRFTABLENAME, vrftablename);
+}
+
 static int bmp_send_initiation(struct bmp *bmp)
 {
 	int len;
-	struct stream *s;
-	s = stream_new(BGP_MAX_PACKET_SIZE);
+	struct stream *s = stream_new(BGP_MAX_PACKET_SIZE);
```

Comment:
is it possible for `stream_new` to return null? is there a check needed here?

Reply:
@ton31337 iirc i was forced to change this by the linter because *need a blank line after variable declarations" even though i didn't write it.
I would rather have this than 3 lines for a single variable initialization.
I admit i can't remember exactly though as that was several months ago. Is this really an issue?

Reply:
> is it possible for `stream_new` to return null? is there a check needed here?

@riw777 i don't think. it looks like stream_new allocates and initializes fields on the allocated value. even if there was an allocation error stream_new could not return NULL without crashing.

https://github.com/FRRouting/frr/blob/c4e030ac87687b15418df652f69e3959c58de0e2/lib/stream.c#L88-L101